packages/dds/tree/src/simple-tree/objectNode.ts

```diff
@@ -173,45 +175,27 @@ function createObjectProxy(
 			const flexNodeSchema = flexNode.schema;
 			assert(flexNodeSchema instanceof FlexObjectNodeSchema, 0x888 /* invalid schema */);
 
-			const flexKey: FieldKey = getFlexKey(viewKey as FieldKey);
-			const flexFieldSchema = flexNodeSchema.objectNodeFields.get(flexKey);
+			const hasField = viewKey in schema.info;
 
-			if (flexFieldSchema === undefined) {
+			if (!hasField) {
 				return allowAdditionalProperties ? Reflect.set(target, viewKey, value) : false;
 			}
+			assert(typeof viewKey === "string", 0x7e1 /* invalid key */);
+			const flexKey: FieldKey | undefined = getFlexKey(viewKey);
 
 			// TODO: Is it safe to assume 'content' is a LazyObjectNode?
 			assert(flexNode instanceof LazyObjectNode, 0x7e0 /* invalid content */);
```

Comment:
I as assuming this was here as a sanity check, but I guess it used to actually be required and isn't now. Removed.

Reply:
Nice. I think it used to be our only import of LazyFlex into simple-tree. However, we've recently added a couple more dependencies on LazyFlex (because, naturally, we already had one, so what's the harm? :D).